ruby/gilded_rose.rb

```diff
@@ -65,4 +24,128 @@ def initialize(name, sell_in, quality)
   def to_s()
     "#{@name}, #{@sell_in}, #{@quality}"
   end
+end
+
+class ItemUpdaterFactory
+  def self.create_updater_for(item)

```

Comment:
This could also be a static method on `Updater` if you want to reduce the number of classes flying around.